### PR TITLE
A few minor fixes + `V2` naming refactor for `modeling` subpackage

### LINKED ISSUE #815: Matplotlib warning `"UserWarning: Starting a Matplotlib GUI outside of the main thread will likely fail."`
We should get rid of this warning.

### PR DESCRIPTION
**Pull request by hmacdope**
## Description
Provide a brief description of the PR's purpose here.
Fixes #815 #783 #714 

## Developers certificate of origin
- [X] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

### CODE REVIEW

**Comment by hmacdope** [comment]
@apayne97 @JenkeScheen @chrisiacovella @ijpulidos can I get a review here please?

File: asapdiscovery-data/asapdiscovery/data/logging.py [hunk 1]
```diff
@@ -66,3 +67,14 @@ def set_level(self, level: int) -> None:
     oechem.OEThrow.SetOutputStream(errfs)
 
 """
+
+
+class HiddenPrint:
```
Comment (kaminow) [inline asapdiscovery-data/asapdiscovery/data/logging.py:72]: love this

File: asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py [hunk 2]
```diff
@@ -571,7 +573,12 @@ def make_logoplot_input(self, resi) -> dict:
             ]
 
         logoplot_base64s_dict = {}
-        with tempfile.TemporaryDirectory() as tmpdirname:
+
+        # hide a shockingly large number of prints from inside logomaker
+        with tempfile.TemporaryDirectory() as tmpdirname, HiddenPrint() as _:
+            import matplotlib
+
+            matplotlib.use("agg")
```
Comment (kaminow) [inline asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py:581]: I'm assuming this prevents some annoying logging from happening in the `logomaker` call below, but just want to make sure this isn't extra code

**Review by kaminow — APPROVED** [review]
looks good! one quick question but I don't think anything needs to change

File: asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py (hunk 2 above)
Comment (hmacdope) [reply asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py:581]: Yep, spot on, stops matplotlib trying to start a GUI and giving a bunch of warnings.

**Pull request merged**